creating short admin password in BCFIPS approved mode gives "Internal server error" page

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

If you try to create a new admin console password with a password under 14 characters you get an "Internal Server Error" page and the following in the logs:

Caused by: org.bouncycastle.crypto.fips.FipsUnapprovedOperationError: password must be at least 112 bits
    at org.bouncycastle.crypto.fips.FipsPBKD$Parameters.<init>(Unknown Source)
    at org.bouncycastle.crypto.fips.FipsPBKD$Parameters.<init>(Unknown Source)
    at org.bouncycastle.crypto.fips.FipsPBKD$ParametersBuilder.using(Unknown Source)
    at org.bouncycastle.jcajce.provider.ProvPBEPBKDF2$BasePBKDF2.engineGenerateSecret(Unknown Source)
    at java.base/javax.crypto.SecretKeyFactory.generateSecret(SecretKeyFactory.java:333)
    at org.keycloak.credential.hash.Pbkdf2PasswordHashProvider.encodedCredential(Pbkdf2PasswordHashProvider.java:118)
    at org.keycloak.credential.hash.Pbkdf2PasswordHashProvider.encodedCredential(Pbkdf2PasswordHashProvider.java:81)
    at org.keycloak.credential.PasswordCredentialProvider.createCredential(PasswordCredentialProvider.java:68)
    ... 29 more

There's two issues with this throwing an Internal Server Error:

  - requires customer to read low level log message to attempt to understand a user input issue

  - this is a bad experience on the first user interaction with out product

 
This is the same issue reported by Marek here: https://github.com/keycloak/keycloak/issues/14314 which was reported fixed.

### Version

24.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

Not "Internal Server Error"

### Actual behavior

"Internal Server Error"

### How to Reproduce?

Steps:
 * Setup Postgresql and FIPS as described in docs.
 * Start Keycloak with BCFIPS in the approved mode with command:
   bin/kc.sh start --fips-mode=strict --features=fips
 * Then go to [http://localhost:8080|http://localhost:8080/] and try to create user {{admin}} with password {{admin}}

 * Get error page:

We are sorry...
An internal server error has occurred

and in logs:

Caused by: org.bouncycastle.crypto.fips.FipsUnapprovedOperationError: password must be at least 112 bits
    at org.bouncycastle.crypto.fips.FipsPBKD$Parameters.<init>(Unknown Source)
    at org.bouncycastle.crypto.fips.FipsPBKD$Parameters.<init>(Unknown Source)
    at org.bouncycastle.crypto.fips.FipsPBKD$ParametersBuilder.using(Unknown Source)
    at org.bouncycastle.jcajce.provider.ProvPBEPBKDF2$BasePBKDF2.engineGenerateSecret(Unknown Source)

### Anything else?

_No response_

--- a/services/src/main/java/org/keycloak/services/resources/WelcomeResource.java
+++ b/services/src/main/java/org/keycloak/services/resources/WelcomeResource.java
@@ -43,6 +43,7 @@ import org.keycloak.cookie.CookieProvider;
 import org.keycloak.cookie.CookieType;
 import org.keycloak.http.HttpRequest;
 import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ModelException;
 import org.keycloak.services.ServicesLogger;
 import org.keycloak.services.managers.ApplianceBootstrap;
 import org.keycloak.services.util.CacheControlUtil;
@@ -133,10 +134,16 @@ public class WelcomeResource {
                 return createWelcomePage(null, "Password and confirmation doesn't match");
             }
 
-            expireCsrfCookie();
+            try {
+                ApplianceBootstrap applianceBootstrap = new ApplianceBootstrap(session);
+                applianceBootstrap.createMasterRealmUser(username, password);
+            } catch (ModelException e) {
+                session.getTransactionManager().rollback();
+                logger.error("Error creating the administrative user", e);
+                return createWelcomePage(null, "Error creating the administrative user: " + e.getMessage());
+            }
 
-            ApplianceBootstrap applianceBootstrap = new ApplianceBootstrap(session);
-            applianceBootstrap.createMasterRealmUser(username, password);
+            expireCsrfCookie();
 
             shouldBootstrap.set(false);
             ServicesLogger.LOGGER.createdTemporaryAdminUser(username);

